crates/symbolicator-js/src/lookup.rs

```diff
@@ -580,12 +584,17 @@ impl ArtifactFetcher {
             self.metrics.record_not_found(SourceFileType::Source);
         }
 
-        // Then fetch the corresponding sourcemap if we have a sourcemap reference
-        let sourcemap_url = match &minified_source.entry {
-            Ok(minified_source) => minified_source.sourcemap_url.as_deref(),
-            Err(_) => None,
+        // Then fetch the corresponding sourcemap reference and debug_id
+        let (sourcemap_url, source_debug_id) = match &minified_source.entry {
+            Ok(minified_source) => (
+                minified_source.sourcemap_url.as_deref(),
+                minified_source.debug_id,
+            ),
+            Err(_) => (None, None),
         };
 
+        let debug_id = debug_id.or(source_debug_id);
```

Comment:
Wasn't 100% sure which `debug_id` you'd want to favour if one is found in the source file.

Reply:
The current version would prefer the debug id that is passed with the request, I would keep it that way.